deployment_scripts/install-infection-monkey-service.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+set -e
+
+SCRIPT_DIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
+SYSTEMD_UNIT_FILENAME="infection-monkey.service"
+SYSTEMD_DIR="/lib/systemd/system"
+MONKEY_BIN="/opt/infection-monkey/bin"
+APPIMAGE_NAME="InfectionMonkey.appimage"
+
+echo_help() {
+  echo "Installs Infection Monkey service to run on boot"
+  echo ""
+  echo "Usage:"
+  echo "    install-infection-monkey-service.sh --user <NAME> --appimage <PATH>"
+  echo "    install-infection-monkey-service.sh --uninstall"
+  echo "    install-infection-monkey-service.sh -h|--help"
+  echo ""
+  echo "Options:"
+  echo "    --user                      User to run the service as"
+  echo "    --appimage               	Path to AppImage"
+  echo "    --uninstall                 Uninstall Infection Monkey service"
+}
+
+install_service() {
+  move_appimage "$2"
+
+  cat > "${SCRIPT_DIR}/${SYSTEMD_UNIT_FILENAME}" << EOF
+[Unit]
+Description=Infection Monkey Runner
+After=network.target
+
+[Service]
+User=$1
+Type=simple
+ExecStart="${MONKEY_BIN}/${APPIMAGE_NAME}"
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+  sudo mv "${SCRIPT_DIR}/${SYSTEMD_UNIT_FILENAME}" "${SYSTEMD_DIR}/${SYSTEMD_UNIT_FILENAME}"
+  sudo systemctl enable "${SYSTEMD_UNIT_FILENAME}" &>/dev/null
+
+  echo -e "The Infection Monkey service has been installed and will start on boot.\n\
+Run 'systemctl start infection-monkey' to start the service now."
+}
+
+uninstall_service() {
+  if [ -f "${MONKEY_BIN}/${APPIMAGE_NAME}" ] ; then
+    sudo rm -f "${MONKEY_BIN}/${APPIMAGE_NAME}"
+  fi
+
+  if [ -f "${SYSTEMD_DIR}/${SYSTEMD_UNIT_FILENAME}" ] ; then
+    sudo systemctl stop "${SYSTEMD_UNIT_FILENAME}" 2>/dev/null
+    sudo systemctl disable "${SYSTEMD_UNIT_FILENAME}" &>/dev/null
+    sudo rm "${SYSTEMD_DIR}/${SYSTEMD_UNIT_FILENAME}"
+    sudo systemctl daemon-reload
+  fi
+
+  echo "The Infection Monkey service has been uninstalled"
+}
+
+user_can_execute() {
+  sudo -u "$1" test -x "$2"
+}
+
+move_appimage() {
+  sudo mkdir -p "${MONKEY_BIN}"
+
+  if [ "$1" != "${MONKEY_BIN}/${APPIMAGE_NAME}" ] ; then
+    sudo cp "$appimage_path" "${MONKEY_BIN}/${APPIMAGE_NAME}"
+  fi
+}
+
+user_exists() {
+  id -u "$1" &>/dev/null
+}
+
+assert_flag() {
```

Comment:
Flags are boolean. `--user` and `--appimage` are parameters which receive arguments.
```suggestion
assert_parameter_supplied() {
```